[Inductor] Track constant's original_fqn mapping

SherlockNoMad opened the pull request:
When compiling an deserialized ExportedProgram, constant’s original_fqn is not populated(). Highlighted line is missing, And a latter assertion is breaking due to original_fqn missing.

```
        constants_info_[0].name = "L__self___w_pre";
	constants_info_[0].dtype = static_cast<int32_t>(cached_torch_dtype_float32);
	constants_info_[0].offset = 0;
	constants_info_[0].data_size = 64;
	constants_info_[0].from_folded = false;
	constants_info_[0].shape = {4, 4};
	constants_info_[0].stride = {4, 1};
	// constants_info_[0].original_fqn = "w_pre";   // this line is missing 
```

Inductor is relying `dynamo_flat_name_to_original_fqn` to populate the original_fqn field. This field originates from `graph_module.meta["dynamo_flat_name_to_original_fqn"]`, and is set during dynamo tracing. However, when compiling 
an deserialized ExportedProgram, we don't do dynamo tracing, thus this field is missing. 

As a fix, I maintain AOTI's own mapping for constant tensor's fqn. 



Differential Revision: D54097073




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames

SherlockNoMad commented on torch/_inductor/compile_fx.py:
```diff
@@ -317,6 +317,7 @@ def fake_tensor_prop(
     The created fake mode will be returned.
     """
     fake_mode = detect_fake_mode(example_inputs)
+    fake_mode.allow_non_fake_inputs = True
```
is it ok to always run with allow_non_fake_inputs?

chenyang78 replied:
Is it because we have mixed fake/non-fake inputs so that we have to set this flat to be True? If so, where do the non-fake inputs come from? Thanks.

SherlockNoMad replied:
Removed this line.

chenyang78 reviewed (approved):
Please fix the linter failure. Otherwise, LGTM. Thanks!

SherlockNoMad committed:
[sigmoid] Switch to use _inductor.aot_compile()

Summary: https://docs.google.com/document/d/1EWFga8czhhWKeovYi-Ws9eJxdfgpBgZN16sE0w_Zgq8/edit

Test Plan: buck2 run -c fbcode.nvcc_arch=a100 'fbcode//mode/dev-nosan' -c fbcode.nvcc_arch=a100 fbcode//sigmoid/inference/test_gpu:e2e_test

Reviewed By: chenyang78

Differential Revision: D54097073

kit1980 commented:
@pytorchbot merge
